framework/src/org/checkerframework/common/value/ValueTransfer.java

```diff
@@ -855,6 +852,78 @@ private Range calculateRangeUnaryOp(
     }
 
     /**
+     * Calculates the result of a binary comparison on a pair of intRange annotations, and refines
+     * annotations appropriately.
+     */
+    private List<Boolean> refineIntRanges(
+            Node leftNode,
+            Node rightNode,
+            ComparisonOperators op,
+            TransferInput<CFValue, CFStore> p,
+            CFStore thenStore,
+            CFStore elseStore) {
+        Range leftRange = getIntRange(leftNode, p);
+        Range rightRange = getIntRange(rightNode, p);
+
+        final Range thenRightRange;
+        final Range thenLeftRange;
+        final Range elseRightRange;
+        final Range elseLeftRange;
+
+        switch (op) {
+            case EQUAL:
+                thenRightRange = rightRange.refineEqualTo(leftRange);
+                thenLeftRange = leftRange.refineEqualTo(rightRange);
+                elseRightRange = rightRange; // Not equals gives almost no information.
+                elseLeftRange = leftRange;
+                break;
+            case GREATER_THAN:
+                thenLeftRange = leftRange.refineGreaterThan(rightRange);
+                thenRightRange = rightRange.refineLessThan(leftRange);
+                elseRightRange = rightRange.refineLessThanEq(leftRange);
+                elseLeftRange = leftRange.refineGreaterThanEq(rightRange);
+                break;
+            case GREATER_THAN_EQ:
+                thenRightRange = rightRange.refineGreaterThanEq(leftRange);
+                thenLeftRange = leftRange.refineLessThanEq(rightRange);
+                elseLeftRange = leftRange.refineLessThan(rightRange);
+                elseRightRange = rightRange.refineGreaterThan(leftRange);
+                break;
+            case LESS_THAN:
+                thenLeftRange = leftRange.refineLessThan(rightRange);
+                thenRightRange = rightRange.refineGreaterThan(leftRange);
+                elseRightRange = rightRange.refineGreaterThanEq(leftRange);
+                elseLeftRange = leftRange.refineLessThanEq(rightRange);
+                break;
+            case LESS_THAN_EQ:
+                thenRightRange = rightRange.refineLessThanEq(leftRange);
+                thenLeftRange = leftRange.refineGreaterThanEq(rightRange);
+                elseLeftRange = leftRange.refineGreaterThan(rightRange);
+                elseRightRange = rightRange.refineLessThan(leftRange);
+                break;
+            case NOT_EQUAL:
+                thenRightRange = rightRange; // Not equals gives almost no information.
+                thenLeftRange = leftRange;
+                elseRightRange = rightRange.refineEqualTo(leftRange);
+                elseLeftRange = leftRange.refineEqualTo(rightRange);
+                break;
+            default:
+                throw new UnsupportedOperationException();
```

Comment:
Can you replace this (here and elsewhere in this class) and the `throw new RuntimeException("Unrecognized conditional operator " + op);` further down, with the proper `ErrorReporter.errorAbort`... that will result in a nicer error message in case this ever happens and isn't much longer to write.